spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/main/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/KubernetesModuleDeployer.java

```diff
@@ -0,0 +1,306 @@
+package org.springframework.cloud.dataflow.module.deployer.kubernetes;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentId;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentRequest;
+import org.springframework.cloud.dataflow.module.ModuleStatus;
+import org.springframework.cloud.dataflow.module.deployer.ModuleArgumentQualifier;
+import org.springframework.cloud.dataflow.module.deployer.ModuleDeployer;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+
+/**
+ * Implements a deployer for the Google Kubernetes project. 
+ * 
+ * @author Florian Rosenberg
+ */
+public class KubernetesModuleDeployer implements ModuleDeployer {
+	
+	protected static final String SCSM_GROUP_KEY = "scsm-group";
+	protected static final String SCSM_LABEL_KEY = "scsm-label";
+	private static final String SCSM_EXTENSION = "scsm-extension";
+	private static final String SCSM_VERSION = "scsm-version";
+	private static final String SCSM_GROUP_ID = "scsm-groupId";
+	private static final String SCSM_ARTIFACT_ID = "scsm-artifactId";
+	private static final String SPRING_MARKER_VALUE = "scsm-module";
+	private static final String SPRING_MARKER_KEY = "role";
+	private static final String PORT_KEY = "port";
+	private static final String SERVER_PORT_KEY = "server.port";
+	private static final String HEALTH_ENDPOINT = "/health";
+	private static final String CONTAINER_NAME = "spring-module-launcher";
+	
+	private static final String SPRING_REDIS_HOST = "SPRING_REDIS_HOST";
+	private static final String SPRING_REDIS_SENTINEL_NODES = "SPRING_REDIS_SENTINEL_NODES";
+	private static final String SPRING_REDIS_SENTINEL_MASTER = "SPRING_REDIS_SENTINEL_MASTER";
+		
+	/** 
+	 * Use the default spring image. Override with --kubernetes.moduleLauncherImage 
+	 * as a Spring Admin parameter to use a different one.
+	 */
+	private static final String DEFAULT_IMAGE_NAME = "springcloud/stream-module-launcher";
```

Comment:
This should not be here, but should be the initialization value in `KubernetesModuleDeployerProperties `